eth/tracers/api_test.go

```diff
@@ -141,24 +134,27 @@ type testBackend struct {
 	refHook func() // Hook is invoked when the requested state is referenced
 	relHook func() // Hook is invoked when the requested state is released
 
-	historical *rpc.Client
+	historical     *rpc.Client
+	mockHistorical *mockHistoricalBackend
 }
 
 // testBackend creates a new test backend. OBS: After test is done, teardown must be
 // invoked in order to release associated resources.
 func newTestBackend(t *testing.T, n int, gspec *core.Genesis, generator func(i int, b *core.BlockGen)) *testBackend {
-	historicalAddr := newMockHistoricalBackend(t)
+	mock := new(mockHistoricalBackend)
+	historicalAddr := newMockHistoricalBackend(t, mock)
```

Comment:
Good improvement to use the mock package, but I think you missed the mock assertion. When setting up a mock, you can add a `defer mock.AssertExpectations(t)` (calling the embedded `Mock`) to see if all calls that we have set up as expected are actually called.

Reply:
Added